### DISCUSSION TITLE
Support for passkeys under cryptoProperties

### DISCUSSION THREAD

**Discussion by prabhu** [CycloneDX/specification#488] [Q&A]
> Based on FIDO standards, passkeys are a replacement for passwords that provide faster, easier, and more secure sign-ins to websites and apps across a user’s devices. Unlike passwords, passkeys are always strong and phishing-resistant.​

Based on my limited understanding, the following are some of the essential properties to catalog the use of passkeys in an organization

- Authenticator transport (usb, nfc, ble, cable, etc)
- Authenticator attachment (removable - platform, cross-platform)
- Resident key requirement
- User verification requirement
- AAGUID - Authenticator Attestation Global Unique ID. Example: Windows Hello (08987058-cadc-4b81-b6e1-30de50dcbe96) and 1Password (bada5566-a7aa-401f-bd96-45619a55120d)
- Whether device-bound or multi-device sync compatible
- Relying party ID - domain name where the passkeys are bound to
- Acceptable public key algorithms - Array of types acceptable by the server
- Attestation conveyance preference (The Relying Party's interest in receiving an attestation statement)
- Attestation format

It would be nice to enhance the [cryptoProperties](https://cyclonedx.org/docs/1.6/json/#components_items_cryptoProperties_assetType) to support passkeys

## References

- https://fidoalliance.org/passkeys/
- https://www.corbado.com/blog/passkeys-javascript
- https://passkeys.dev/docs/intro/what-are-passkeys/
- https://github.com/duo-labs/py_webauthn/tree/master
- https://github.com/duo-labs/py_webauthn/blob/3c04e4831086129be45be54fd144d4ed137075fa/webauthn/helpers/structs.py#L248
- https://www.w3.org/TR/webauthn-2/#enum-transport

**Comment by stevespringett** [comment]
cc: @bhess
